azure-client-runtime/src/main/java/com/microsoft/azure/AzureServiceClient.java

```diff
@@ -49,6 +56,20 @@ protected AzureServiceClient(String baseUrl, ServiceClientCredentials credential
      */
     protected AzureServiceClient(RestClient restClient) {
         super(restClient);
+        rpHttpClient = new RxNettyClient(new ChannelHandlerConfig(new Func0<ChannelHandler>() {
+            @Override
+            public ChannelHandler call() {
+                return new AzureTokenCredentialsHandler((AzureTokenCredentials) AzureServiceClient.this.restClient().credentials());
```

Comment:
Is there someway to get to credentials() without using restClient()?

Reply:
Also, will we create a new AzureTokenCredentialsHandler for each outgoing HTTP request? If so, that should be pulled out into a member variable of AzureServiceClient.

Reply:
Let's figure something out for credentials. Probably one of the more hacked up parts of this.

Reply:
Yeah, the handler is getting re-instantiated but doesn't really have any of its own state beyond the credentials which are already shared. It could be a final local or an instance variable, I think.